### ISSUE TITLE
kubectl cluster-info dump: error: a versioned object must be passed to a printer

### ISSUE THREAD

**Issue by DizzyThermal** [kubernetes/kubectl#586]
**Background**:
I am performing prerequisite steps on my OpenShift Origin (OKD) v3.11 cluster to prepare to deploy [Ambassador](https://www.getambassador.io). The documentation I'm following [to deploy is here](https://www.getambassador.io/user-guide/getting-started/), specifically the part where it asks me to check if RBAC is enabled by running:
```
$ kubectl cluster-info dump --namespace kube-system | grep authorization-mode
```

this results in the error:
```
error: a versioned object must be passed to a printer
```

I understand that the answer to my actual question - whether RBAC is enabled in my OKD cluster - is likely **Yes**, but I still would expect the dump to return content, not this error.

I've searched keywords, such as: **kubectl cluster-info dump versioned object printer**, but I haven't come across a discussion about this and wasn't sure if this was a bug or something I'm failing to do.

**Kubernetes version** (use `kubectl version`): 
```
Client Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.0+d4cacc0", GitCommit:"d4cacc0", GitTreeState:"clean", BuildDate:"2018-10-15T09:45:30Z", GoVersion:"go1.10.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.0+d4cacc0", GitCommit:"d4cacc0", GitTreeState:"clean", BuildDate:"2019-02-01T22:34:55Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Cloud provider or hardware configuration**: OpenShift Origin (OKD) v3.11
- **OS** (e.g. from /etc/os-release): RHEL Server 7.6 (Maipo)
- **Kernel** (e.g. `uname -a`): 3.10.0-957.1.3.el7.x86_64

--
Are there additional options or flags that need to be specified to make `kubectl cluster-info dump`  work?

Thanks in advance

**Comment by DizzyThermal** [comment]
@juanvallejo of course, here's that output:
```
[root@ns1 ~]# kubectl cluster-info dump --v=8 --namespace=kube-system
I0207 18:21:29.577326   12797 loader.go:359] Config loaded from file /root/.kube/config
I0207 18:21:29.578137   12797 loader.go:359] Config loaded from file /root/.kube/config
I0207 18:21:29.578274   12797 round_trippers.go:383] GET https://console.example.com:8443/api/v1/nodes
I0207 18:21:29.578291   12797 round_trippers.go:390] Request Headers:
I0207 18:21:29.578301   12797 round_trippers.go:393]     Accept: application/json, */*
I0207 18:21:29.578312   12797 round_trippers.go:393]     User-Agent: kubectl/v1.11.0+d4cacc0 (linux/amd64) kubernetes/d4cacc0
I0207 18:21:29.578322   12797 round_trippers.go:393]     Authorization: Bearer <snip>
I0207 18:21:29.598701   12797 round_trippers.go:408] Response Status: 200 OK in 20 milliseconds
I0207 18:21:29.598803   12797 round_trippers.go:411] Response Headers:
I0207 18:21:29.598824   12797 round_trippers.go:414]     Content-Type: application/json
I0207 18:21:29.598836   12797 round_trippers.go:414]     Date: Thu, 07 Feb 2019 23:21:29 GMT
I0207 18:21:29.598852   12797 round_trippers.go:414]     Cache-Control: no-store
I0207 18:21:29.599230   12797 request.go:897] Response Body: {"kind":"NodeList","apiVersion":"v1","metadata":{"selfLink":"/api/v1/nodes","resourceVersion":"930670"},"items":[{"metadata":{"name":"glusterfs1.example.com","selfLink":"/api/v1/nodes/glusterfs1.example.com","uid":"6458bbdb-28a4-11e9-9502-005056b9957f","resourceVersion":"930639","creationTimestamp":"2019-02-04T17:43:33Z","labels":{"beta.kubernetes.io/arch":"amd64","beta.kubernetes.io/os":"linux","glusterfs":"storage-host","kubernetes.io/hostname":"glusterfs1.example.com","node-role.kubernetes.io/compute":"true"},"annotations":{"node.openshift.io/md5sum":"99b90b4178a1448ad104d2e18a26d486","volumes.kubernetes.io/controller-managed-attach-detach":"true"}},"spec":{"unschedulable":true},"status":{"capacity":{"cpu":"8","hugepages-1Gi":"0","hugepages-2Mi":"0","memory":"16266124Ki","pods":"250"},"allocatable":{"cpu":"8","hugepages-1Gi":"0","hugepages-2Mi":"0","memory":"16163724Ki","pods":"250"},"conditions":[{"type":"OutOfDisk","status":"False","lastHeartbeatTime":"2019-02-07T23:21:19Z","lastTransitionTime":"2019-02-04T17:43:33Z [truncated 48690 chars]
F0207 18:21:29.601834   12797 helpers.go:119] error: a versioned object must be passed to a printer
```

Thanks for following up!

**Comment by akowasch** [comment]
The bug seems to have been fixed in Kubernetes Client Version v1.11.3 and later.
